app/controllers/subject_controller.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# Handling main subject funcionality: basic CRUD (create, delete, update subject and it's description)
+class SubjectController < ApplicationController
+  before_action :check_access, only: [:index, :show, :edit, :update, :destroy, :display_all, :create, :new]
+  before_action :find_user, only: [:show, :edit, :update, :destroy]
+
+  # Displays all users subjects
+  def index
+    # Gets current logged in user subjects
+    @subject = current_user.subjects
```

Comment:
instance variable `@subject` name is misleading. Name indicates that it contains single subject, but it actually contains list of subjects. DL;DR: replace `@subject` with `@subjects`

Reply:
fixed, obvious mistake, my bad